rllib/algorithms/tests/test_algorithm_config.py

```diff
@@ -185,6 +189,225 @@ class A:
+                    ),
+                )
+
+        ########################################
+        config = SingleAgentAlgoConfig().rl_module(_enable_rl_module_api=True)
+        config.validate()
+
+        spec, expected = self._get_expected_marl_spec(config, DiscreteBCTorchModule)
+        self._assertEqualMARLSpecs(spec, expected)
+
+        # expected module should become the passed module if we pass it in.
+        spec, expected = self._get_expected_marl_spec(
+            config, CustomRLModule2, passed_module_class=CustomRLModule2
+        )
+        self._assertEqualMARLSpecs(spec, expected)
+
+        ########################################
+        config = SingleAgentAlgoConfig().rl_module(
+            _enable_rl_module_api=True,
+            rl_module_spec=MultiAgentRLModuleSpec(
+                module_specs={
+                    "p1": SingleAgentRLModuleSpec(module_class=CustomRLModule1),
+                    "p2": SingleAgentRLModuleSpec(module_class=CustomRLModule1),
+                }
+            ),
+        )
+        config.validate()
+
+        spec, expected = self._get_expected_marl_spec(config, CustomRLModule1)
+        self._assertEqualMARLSpecs(spec, expected)
+
+        ########################################
+        # We want to override the default rl module
+        config = SingleAgentAlgoConfig().rl_module(
+            _enable_rl_module_api=True,
+            rl_module_spec=SingleAgentRLModuleSpec(module_class=CustomRLModule1),
+        )
+        config.validate()
+
+        spec, expected = self._get_expected_marl_spec(config, CustomRLModule1)
+        self._assertEqualMARLSpecs(spec, expected)
+
+        # expected module should become the passed module if we pass it in.
+        spec, expected = self._get_expected_marl_spec(
+            config, CustomRLModule2, passed_module_class=CustomRLModule2
+        )
+        self._assertEqualMARLSpecs(spec, expected)
+        ########################################
+        # We now want to tell the algorithm to use a different rl module for each
+        # RLModule
+        config = SingleAgentAlgoConfig().rl_module(
+            _enable_rl_module_api=True,
+            rl_module_spec=MultiAgentRLModuleSpec(
+                module_specs=SingleAgentRLModuleSpec(module_class=CustomRLModule1)
+            ),
+        )
+        config.validate()
+
+        spec, expected = self._get_expected_marl_spec(config, CustomRLModule1)
+        self._assertEqualMARLSpecs(spec, expected)
+
+        # expected module should become the passed module if we pass it in.
+        spec, expected = self._get_expected_marl_spec(
+            config, CustomRLModule2, passed_module_class=CustomRLModule2
+        )
+        self._assertEqualMARLSpecs(spec, expected)
+
+        ########################################
+        # We now want to tell the algorithm to use a heterogeneous rl module for each
+        # RLModule
```

Comment:
When you say heterogeneous here, do you mean policies will be learnt independently? Because the specs are homogeneous.

Reply:
Clarified all the explanations. Hope these new wordings are more clear about what is happening under each test-case.